redpanda/templates/configmap.yaml

```diff
@@ -37,7 +37,7 @@ metadata:
     {{- toYaml . | nindent 4 }}
 {{- end }}
 data:
-{{- if (include "redpanda.semver" . | semverCompare ">=22.1.1") }}
+{{- if or (eq .Values.image.repository "vectorized/redpanda-nightly") (include "redpanda.semver" . | semverCompare ">=22.1.1") }}
```

Comment:
Let's go the other direction and only check it if it's `vectorized/redpanda`. That would allow developers to host their own images for testing as well.